alacritty/src/display/hint.rs

```diff
@@ -204,7 +204,8 @@ pub struct HintMatch {
 impl HintMatch {
     #[inline]
     pub fn should_highlight(&self, point: Point, pointed_hyperlink: Option<&Hyperlink>) -> bool {
-        self.bounds.contains(&point) && self.hyperlink.as_ref() == pointed_hyperlink
+        self.hyperlink.as_ref() == pointed_hyperlink
+            && (self.hyperlink.is_some() || self.bounds.contains(&point))
```

Comment:
What the reason to check `is_some`, I think it's already covered?

Reply:
If it's a hyperlink escape, we ALWAYS highlight regardless of the bounds. If we don't check for `is_some`, we'd only highlight with in bounds.